src/RolloutScorer/RolloutScorerAzureFunction/RolloutScorerFunction.cs

```diff
@@ -1,135 +1,187 @@
-            IEnumerable<AnnotationEntity> relevantDeployments =
-                deploymentEntries.Where(d => (d.Ended ?? DateTimeOffset.MaxValue) > scorecardEntries.Last().Date.AddDays(ScoringBufferInDays));
-            log.LogInformation($"INFO: Found {relevantDeployments?.Count() ?? -1} relevant deployments (deployments which occurred " +
-                $"after the last scorecard). (-1 indicates that null was returned.)");
+
+            // TODO: Change this line to fix the timing. Logic should be based on last day of a rollout (how do we determine what the last day of a rollout is?) rather than the last time it was scored. 
+            IEnumerable<AnnotationEntity> relevantDeployments = deploymentEntries.Where(d => (d.Ended ?? DateTimeOffset.MaxValue) > scorecardEntries.Last().Date.AddDays(ScoringBufferInDays));
+            _logger.LogInformation($"INFO: Found {relevantDeployments?.Count() ?? -1} relevant deployments (deployments which occurred " + $"after the last scorecard). (-1 indicates that null was returned.)");
 
             if (relevantDeployments.Count() > 0)
             {
-                log.LogInformation("INFO: Checking to see if the most recent deployment occurred more than two days ago...");
+                _logger.LogInformation("INFO: Checking to see if the most recent deployment occurred more than two days ago...");
                 // We have only want to score if the buffer period has elapsed since the last deployment
                 if ((relevantDeployments.Last().Ended ?? DateTimeOffset.MaxValue) < DateTimeOffset.UtcNow - TimeSpan.FromDays(ScoringBufferInDays))
                 {
                     var scorecards = new List<Scorecard>();
 
-                    log.LogInformation("INFO: Rollouts will be scored. Fetching GitHub PAT...");
-                    SecretBundle githubPat;
-                    using (KeyVaultClient kv = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(tokenProvider.KeyVaultTokenCallback)))
-                    {
-                        githubPat = await kv.GetSecretAsync(Utilities.KeyVaultUri, Utilities.GitHubPatSecretName);
-                    }
-
+                    _logger.LogInformation("INFO: Rollouts will be scored. Fetching GitHub PAT...");
+                    SecretBundle githubPat = await _keyVaultClient.GetSecretAsync(_keyVaultOptions.Uri, _gitHubOptions.PatSecretName);
+                    
                     // We'll score the deployments by service
                     foreach (var deploymentGroup in relevantDeployments.GroupBy(d => d.Service))
                     {
-                        log.LogInformation($"INFO: Scoring {deploymentGroup?.Count() ?? -1} rollouts for repo '{deploymentGroup.Key}'");
+                        string currentRepo = deploymentGroup.Key;
+
+                        _logger.LogInformation($"INFO: Scoring {deploymentGroup?.Count() ?? -1} rollouts for repo '{deploymentGroup.Key}'");
```

Comment:
Little weird to logINFORMATION and then start the statement with "INFO", but I guess you didn't do that. :-)

Reply:
My assumption is when @jonfortescue first made these log statements they were console.writeline's and when he added logger, he kept the text and changed just the mechanism :)

Reply:
Yeah, I think the logging is definitely something that needs to be refactored. I mentioned to @jonfortescue that this project should have a custom logger if he needs all the fancy coloring or whatever the Utilities class is providing.